pkg/auth/dependency/newinteraction/nodes/do_generate_recovery_code.go

```diff
@@ -0,0 +1,43 @@
+package nodes
+
+import (
+	"github.com/authgear/authgear-server/pkg/auth/dependency/newinteraction"
+)
+
+func init() {
+	newinteraction.RegisterNode(&NodeDoGenerateRecoveryCode{})
+}
+
+type EdgeDoGenerateRecoveryCode struct {
+	RecoveryCodes []string
+}
+
+func (e *EdgeDoGenerateRecoveryCode) Instantiate(ctx *newinteraction.Context, graph *newinteraction.Graph, rawInput interface{}) (newinteraction.Node, error) {
+	return &NodeDoGenerateRecoveryCode{
+		RecoveryCodes: e.RecoveryCodes,
+	}, nil
+}
+
+type NodeDoGenerateRecoveryCode struct {
+	RecoveryCodes []string `json:"recovery_nodes"`
+}
+
+func (n *NodeDoGenerateRecoveryCode) Apply(perform func(eff newinteraction.Effect) error, graph *newinteraction.Graph) error {
+	err := perform(newinteraction.EffectRun(func(ctx *newinteraction.Context) error {
+		if len(n.RecoveryCodes) > 0 {
```

Comment:
Personally I'd like to avoid conditionally update the database in Do nodes (i.e. Do nodes always do something, otherwise it is skipped), but it does seems cleaner to just enter the Do nodes unconditionally.

What do you think about this? If you think it's better I'll update `DoVerifyIdentity` to use the same way.